src/main/java/org/musicbrainz/search/solrwriter/MBXMLWriter.java

```diff
@@ -321,11 +333,35 @@ private static void adjustScore(float maxScore, Object object,
 
 	public void write(Writer writer, SolrQueryRequest req, SolrQueryResponse res)
 			throws IOException {
+
+		StringWriter sw = new StringWriter();
+
+		Exception solrError = res.getException();
+
+		if (solrError != null)
+		{
+			MBError mbError = new MBError();
+			int code = Integer.parseInt(res.getResponseHeader().get("status").toString());
+
+			mbError.setCode(code);
```

Comment:
I'm not sure if this is OK with JAXB, but maybe add a constructor to MBError that takes 2 arguments and sets code and message directly?

Reply:
I tried that, doesn't seem to work :(